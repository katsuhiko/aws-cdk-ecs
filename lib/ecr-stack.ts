import { Construct, Stack, StackProps } from '@aws-cdk/core';
import { Repository } from '@aws-cdk/aws-ecr';

export class EcrStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    new Repository(this, 'WebRepo', {
      repositoryName: 'demo-cdk-web',
      imageScanOnPush: true,
      lifecycleRules: [
        { description: 'Delete old images', maxImageCount: 5 },
      ],
    });

    new Repository(this, 'AppRepo', {
      repositoryName: 'demo-cdk-app',
      imageScanOnPush: true,
      lifecycleRules: [
        { description: 'Delete old images', maxImageCount: 5 },
      ],
    });
  }
}
