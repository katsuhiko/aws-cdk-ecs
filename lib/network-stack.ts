import { Construct, Stack, StackProps } from '@aws-cdk/core';
import { SubnetType, Vpc } from '@aws-cdk/aws-ec2';

export class NetworkStack extends Stack {

  public readonly vpc: Vpc

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    this.vpc = new Vpc(this, 'VPC', {
      maxAzs: 2,
      subnetConfiguration: [
        { name: 'Frontend', subnetType: SubnetType.PUBLIC, cidrMask: 24 },
        { name: 'Application', subnetType: SubnetType.PUBLIC, cidrMask: 24 },
        { name: 'Datastore', subnetType: SubnetType.ISOLATED, cidrMask: 24 },
      ],
    });
  }
}
