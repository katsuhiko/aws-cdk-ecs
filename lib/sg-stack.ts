import { Construct, Stack, StackProps, Tags } from '@aws-cdk/core';
import { IVpc, Peer, Port, SecurityGroup } from '@aws-cdk/aws-ec2';

interface SgStackProps extends StackProps {
  vpc: IVpc
}

export class SgStack extends Stack {

  public readonly bastionSg: SecurityGroup;
  public readonly albSg: SecurityGroup;
  public readonly webSg: SecurityGroup;
  public readonly rdsSg: SecurityGroup;

  constructor(scope: Construct, id: string, props: SgStackProps) {
    super(scope, id, props);

    this.bastionSg = new SecurityGroup(this, 'BastionSg', {
      vpc: props.vpc,
    });
    Tags.of(this.bastionSg).add('Name', 'SgStack/BastionSg');

    this.albSg = new SecurityGroup(this, 'AlbSg', {
      vpc: props.vpc,
    });
    Tags.of(this.albSg).add('Name', 'SgStack/AlbSg');
    this.albSg.addIngressRule(Peer.anyIpv4(), Port.tcp(80));
    this.albSg.addIngressRule(Peer.anyIpv4(), Port.tcp(443));

    this.webSg = new SecurityGroup(this, 'WebSg', {
      vpc: props.vpc,
    });
    Tags.of(this.webSg).add('Name', 'SgStack/WebSg');
    this.webSg.addIngressRule(this.albSg, Port.tcp(80));

    this.rdsSg = new SecurityGroup(this, 'RdsSg', {
      vpc: props.vpc,
    });
    Tags.of(this.rdsSg).add('Name', 'SgStack/RdsSg');
    this.rdsSg.addIngressRule(this.bastionSg, Port.tcp(3306));
    this.rdsSg.addIngressRule(this.webSg, Port.tcp(3306));
  }
}
