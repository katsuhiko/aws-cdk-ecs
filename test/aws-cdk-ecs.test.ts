import { expect as expectCDK, matchTemplate, MatchStyle } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as AwsCdkEcs from '../lib/aws-cdk-ecs-stack';

test('Empty Stack', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new AwsCdkEcs.AwsCdkEcsStack(app, 'MyTestStack');
    // THEN
    expectCDK(stack).to(matchTemplate({
      "Resources": {}
    }, MatchStyle.EXACT))
});
