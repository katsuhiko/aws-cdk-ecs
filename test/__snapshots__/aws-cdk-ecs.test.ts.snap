// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Stack 1`] = `Object {}`;
